Add relocate_images to build.sh

deeglaze opened the pull request:
This addition to build.sh interprets a markdown image directive as
alternative text and relocates the source image to a destination path.
The alternative text may have the initial description and the stated
destination values interpolated into the replacement text as
$description and $destination.

Documentation that references images may now have their dependencies
forwarded to the right website location.

The downside to this change is that each <!-- ... --> comment must be single-line. This can mean very long lines in sources. I haven't yet found this limitation to be too burdensome.

This has been tested on the typical build workflow with image relocation
directives added to api-overview.md and overview.md.

KeithMoyer commented on scripts/build.sh:
```diff
@@ -99,6 +99,75 @@ if [[ -f "${ASYLO_LOCATION}" ]]; then
+# <!--asylo:image-replace-begin(dest)-->
+#
+# ![description](source)
+# <!--asylo:image-replace-with replacment[$destination, $description free]-->
```
replacment typo

deeglaze replied:
Fixed.

KeithMoyer commented on scripts/build.sh:
```diff
@@ -99,6 +99,75 @@ if [[ -f "${ASYLO_LOCATION}" ]]; then
+    git_add_command=$(printf '"pushd \\"%s\\"; git add %s; popd"' "${SITE}" "${full_destination}")
+    git_add_destination="system(${git_add_command})"
+  fi
+  awk -i inplace "{
```
produces error: "awk: not an option: -i"

Perhaps you meant gawk? If so, that would need to be added to the Dockerfile.

deeglaze replied:
Changed to use temp file and mv.

KeithMoyer commented on scripts/build.sh:
```diff
@@ -99,6 +99,75 @@ if [[ -f "${ASYLO_LOCATION}" ]]; then
+#
+# becomes
+#
+# replacement{$destination := dest, $description := description}
```
It's unclear what parts of this come from where (e.g., is "replacement" arbitrary text taken from the replace-with line or some special command, does $description or description relate to $description above (both?)), possibly because I am not following what the resulting text even means. Can you describe what this resulting text is doing? Is "replacement" some special command (if so, is there a link to docs?)

deeglaze replied:
Added a section to README.md and reworded a bit.

KeithMoyer replied:
Thanks.

KeithMoyer replied:
I saw it momentarily, but it seems your force push removed it?

KeithMoyer reviewed (changes requested):
I'm unsure what problem(s) this is solving, potentially because I'm not understanding the "Documentation that references images may now have their dependencies forwarded to the right website location." statement.

Could you clarify? Does jekyll not fully support the normal markdown alt text syntax? In what type of case is file relocation desirable?

deeglaze committed:
Add relocate_images to build.sh

This addition to build.sh interprets a markdown image directive as
alternative text and relocates the source image to a destination path.
The alternative text may have the initial description and the stated
destination values interpolated into the replacement text as
$description and $destination.

Documentation that references images may now have their dependencies
forwarded to the right website location.

KeithMoyer commented on scripts/build.sh:
```diff
@@ -99,6 +99,82 @@ if [[ -f "${ASYLO_LOCATION}" ]]; then
+  local transformed=$(mktemp)
+  awk "{
+  if (\$0 ~ ${begin_re}) {
+    destination = gensub(${begin_re}, \"\\\\1\", 1, \$0);
```
awk: line 36: function gensub never defined

deeglaze replied:
Added gawk to the container.

KeithMoyer replied:
Thanks. After rebuilding the container image, I can confirm these build.sh modifications result in no changes to the website.

deeglaze committed:
Add gawk to the website build container
